### PR TITLE
Bulk insert inventory units for for processing large orders.

### PR DESCRIPTION
**Pull request by nimish-mehta**
1. Rewrite the inventory unit creation process to perform bulk inserts.
2. Do array pushes instead of append and re assign
3. Preload data wherever possible instead of accessing in loops.

### CODE REVIEW

File: core/app/models/spree/order.rb [hunk 1]
```diff
@@ -404,7 +404,7 @@ def available_payment_methods
     end
 
     def insufficient_stock_lines
-      line_items.select(&:insufficient_stock?)
+      line_items.includes(variant: :product).select(&:insufficient_stock?)
```
Comment (tanmay3011) [inline core/app/models/spree/order.rb:407]: @nimish-mehta I think we need to move this at the controller level. Or check whether it's already included or not.

File: core/spec/models/spree/order_spec.rb [hunk 2]
```diff
@@ -222,8 +222,12 @@ def compute(computable)
```
Comment (tanmay3011) [inline core/spec/models/spree/order_spec.rb:222]: We also need to add a test case for bulk insert of inventory units

File: core/app/models/spree/shipment.rb [hunk 3]
```diff
@@ -241,6 +240,20 @@ def selected_shipping_rate_id=(id)
...
+      _time = Time.zone.now.to_s(:db)
+      insert_clause  = %Q{ INSERT INTO #{Spree::InventoryUnit.quoted_table_name} (#{conn.quote_column_name("state")}, #{conn.quote_column_name("variant_id")}, #{conn.quote_column_name("order_id")}, #{conn.quote_column_name("shipment_id")}, #{conn.quote_column_name("created_at")}, #{conn.quote_column_name("updated_at")}, #{conn.quote_column_name("line_item_id")}) VALUES }
+      inventory_list.group_by(&:state).each do |state, inventory_list_items|
+        inventory_list_items.group_by(&:variant_id).each do |variant_id, items|
+          values = items.map do |item|
+            %Q{ ('#{ state }', #{ variant_id }, #{ self.order_id }, #{ self.id }, '#{ _time }', '#{ _time }', #{ item.line_item_id }) }
+          end.join(", ")
+          conn.execute(insert_clause + values)
+        end
+      end
```
Comment (priyank-gupta) [inline core/app/models/spree/shipment.rb:254]: @nimish-mehta We should reload `inventory_units` at the end of this method to ensure that we get inserted units.

**Comment by nimish-mehta** [comment]
see #164

**Pull request closed without merging**